test/yaml_edit/test_case.dart

```diff
@@ -0,0 +1,318 @@
+    /// the input file.
+    final goldenOutput = states.join('---$inputLineEndings');
+
+    goldenFile.writeAsStringSync(goldenOutput);
+    state = TestCaseStates.createdGoldenFile;
+  }
+
+  /// Tests the golden file. Ensures that the number of states are the same, and
+  /// that the individual states are the same.
+  void testGoldenFile(File goldenFile) {
+    var inputFileName = inputUri.toFilePath(windows: false).split('/').last;
+    List<String> goldenStates;
+    var golden = goldenFile.readAsStringSync();
+
+    if (detectWindowsLineEndings(golden)) {
+      goldenStates = golden.split('---\r\n');
+    } else {
+      goldenStates = golden.split('---\n');
+    }
+
+    group('testing $inputFileName - input and golden files have', () {
+      test('same number of states', () {
+        expect(states.length, equals(goldenStates.length));
+      });
+
+      for (var i = 0; i < states.length; i++) {
+        test('same state $i', () {
+          expect(states[i], equals(goldenStates[i]));
+        });
+      }
+    });
+
+    state = TestCaseStates.testedGoldenFile;
+  }
+}
+
+/// Converts a [YamlList] into a Dart list.
+List getValueFromYamlList(YamlList node) {
+  return node.value.map((n) {
+    if (n is YamlNode) return getValueFromYamlNode(n);
+    return n;
+  }).toList();
+}
+
+/// Converts a [YamlMap] into a Dart Map.
+Map getValueFromYamlMap(YamlMap node) {
+  var keys = node.keys;
+  var result = {};
+  for (var key in keys) {
+    result[key.value] = result[key].value;
+  }
+
+  return result;
+}
+
+/// Converts a [YamlNode] into a Dart object.
+dynamic getValueFromYamlNode(YamlNode node) {
+  switch (node.runtimeType) {
+    case YamlList:
+      return getValueFromYamlList(node);
+    case YamlMap:
+      return getValueFromYamlMap(node);
+    default:
+      return node.value;
+  }
+}
+
+/// Converts the list of modifications from the raw input to [YamlModification]
+/// objects.
+List<YamlModification> parseModifications(List<dynamic> modifications) {
+  return modifications.map((mod) {
+    Object value;
+    int index;
+    int deleteCount;
+    final method = getModificationMethod(mod[0] as String);
+
+    final path = mod[1] as List;
+
+    if (method == YamlModificationMethod.appendTo ||
+        method == YamlModificationMethod.update ||
+        method == YamlModificationMethod.prependTo) {
+      value = mod[2];
+    } else if (method == YamlModificationMethod.insert) {
+      index = mod[2];
+      value = mod[3];
+    } else if (method == YamlModificationMethod.splice) {
+      index = mod[2];
+      deleteCount = mod[3];
+
+      if (mod[4] is! List) {
+        throw ArgumentError('Invalid array ${mod[4]} used in splice');
+      }
+
+      value = mod[4];
+    }
+
+    return YamlModification(method, path, index, value, deleteCount);
+  }).toList();
+}
+
+/// Gets the YAML modification method corresponding to [method]
+YamlModificationMethod getModificationMethod(String method) {
```

Comment:
Do these need to be exported? Let's review what we need to export from this file, and hide as much as reasonably possible.

Reply:
Updated the file to only export the `TestCases` class!